src/commonMain/kotlin/fr/acinq/lightning/channel/Channel.kt

```diff
@@ -283,6 +234,129 @@ sealed class ChannelState {
         else -> actions
     }
 
+    /** Add actions for some transitions */
+    private fun ChannelContext.onTransition(newState: ChannelState): List<ChannelAction> {
+        val oldState = when (this@ChannelState) {
+            is Offline -> this@ChannelState.state
+            is Syncing -> this@ChannelState.state
+            else -> this@ChannelState
+        }
+        return when {
+            // we only want to fire the PaymentSent event when we transition to Closing for the first time
+            oldState is ChannelStateWithCommitments && oldState !is WaitForInit && oldState !is Closing && newState is Closing -> emitClosingEvents(oldState, newState)
```

Comment:
Since we check that `oldState` has commitments, it cannot be a `WaitForInit`:

```suggestion
            oldState is ChannelStateWithCommitments && oldState !is Closing && newState is Closing -> emitClosingEvents(oldState, newState)
```

Reply:
Fixed in https://github.com/ACINQ/lightning-kmp/pull/458/commits/e47435021275f95ffbcf6a6a32e85405abf6b65b

Reply:
`WaitForInit` isn't a `ChannelStateWithCommitments` though.